### DISCUSSION TITLE
Clover doesn't see NVMe

### DISCUSSION THREAD

**Discussion by Uxorious** [CloverHackyColor/CloverBootloader#375] [Q&A]
Trying to get an old server (Tyan S7012 motherboard) to work with a PCIe NVMe adapter, but so far no luck.
Managed to boot the Windows installer on USB from Clover, so Windows could at least install to NVMe (normally it would complain that the BIOS doesn't support booting from it).
But Clover then never sees the Windows install on the NVMe on subsequent boots.

From digging into logs, I see that it *DID* see the WD SN750 NVMe card during boot:
`1:486  0:000  PCI (00|05:00.00) : 15B7 5006 class=010802`

It also loaded the NVMe driver:
`1:517  0:000  Loading NvmExpressDxe.efi  status=Success`

But it doesn't see any block devices on it:
```
6:313  0:006  === [ ScanVolumes ] =============================
6:313  0:000  Found 2 volumes with blockIO
6:313  0:000  - [00]: Volume: PcieRoot(0x0)\Pci(0x9,0x0)\Pci(0x0,0x0)\USB(0x80,0x0)
6:315  0:001          Result of bootcode detection: bootable unknown (legacy)
6:315  0:000  - [01]: Volume: PcieRoot(0x0)\Pci(0x9,0x0)\Pci(0x0,0x0)\USB(0x80,0x0)\HD(1,MBR,0x0D4289F4,0x3F,0xECDFC1)
6:317  0:001          Result of bootcode detection: bootable unknown (legacy)
6:323  0:005            label : BDU
6:323  0:000          This is SelfVolume !!
6:323  0:000  Searching volumes for latest nvram.plist ...
```

I am using boot7 because boot6 causes my mouse/keyboard to not work.
(Another issue that I might file as a separate bug when I've investigated more ... boot7 resets right after the 7 shows on the screen if I have a SATA CD drive connected)

Any ideas why it's not seeing the NVMe drive?

**Comment by Uxorious** [comment]
I was also trying to enable the debug log (Boot/Debug -> true in the config file), but no debug log is generated (and boot is as fast as it always was). I know it's reading the config file, because I added some stuff to the Arguments as well, and those I see in the bdmesg.efi log...

**Reply by serdeliuk** [reply]
Probably you need to install efi drivers for your partitions, for example fat32.efi or ntfs.efi drivers. And be sure the scan > legacy is true.
